openapi/openapi.yaml

```diff
@@ -0,0 +1,616 @@
+openapi: 3.0.0
+
+info:
+  version: 0.0.1
+  title: Blog Service
+  description: |
+    Blog for the OnHomeProject project by SoftServe
+
+
+tags:
+  - name: "users"
+  - name: "posts"
+  - name: "comments"
+  - name: "tags"
+
+paths:
+
+  /users:
+    post:
+      summary: Sign up a new user
+      tags:
+        - "users"
+      operationId: signUp
+      requestBody:
+        description: object User
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/User"
+      responses:
+        "201":
+          description: CREATED
+        "400":
+          description: BAD REQUEST
+
+
+    get:
+      summary: This gets all users
+      description: Return all users with paggination, could be sorted
+      tags:
+        - "users"
+      operationId: getAllUsers
+      parameters:
+        - in: query
+          name: sort
+          schema:
+            type: string
+            enum: [ asc, desc ]
```

Comment:
Consider the following      
     - in: query
          name: id
          schema:

      - in: query
          name: name
          schema:

      - in: query
          name: post_count
          schema:


?sort=-id
?search=id bt 5 - 50

/users?id=5    == /users/5

SELECT * from users LIMIT 50 OFFSET  49950 where name like '%Vasya%' ORDER BY id;

`users?pageNum=999&pageSize=50`

`X-Total-Count`

`users?pageNum=1&pageSize=50`
`X-Total-Count=500`